src/main/scala/com/codacy/client/stash/client/StashClient.scala

```diff
@@ -1,24 +1,23 @@
 package com.codacy.client.stash.client
 
-import com.codacy.client.stash.client.auth.WSSignatureCalculatorRSA
+import com.codacy.client.stash.client.auth.{Authenticator, OAuth1Authenticator}
 import com.codacy.client.stash.util.HTTPStatusCodes
-import com.ning.http.client.oauth.{ConsumerKey, RequestToken}
 import com.ning.http.client.{AsyncHttpClient, AsyncHttpClientConfigBean}
 import play.api.http.{ContentTypeOf, Writeable}
 import play.api.libs.json._
+import play.api.libs.ws.WSRequest
 import play.api.libs.ws.ning.NingWSClient
 
 import scala.concurrent.Await
 import scala.concurrent.duration._
 
-class StashClient(baseUrl: String, key: String, secretKey: String, token: String, secretToken: String, acceptAllCertificates: Boolean = false) {
+class StashClient(baseUrl: String, authenticator: Option[Authenticator] = None, acceptAllCertificates: Boolean = false) {
+  @deprecated
```

Comment:
Also please add a message to deprecated since it is complaining when compiling with `@deprecated now takes two arguments; see the scaladoc.`. Just a quick tip on how to use the new version.

Reply:
done